third-party/realdds/src/dds-device-server.cpp

```diff
@@ -135,3 +172,33 @@ void dds_device_server::publish_notification( topics::flexible_msg && notificati
     _notification_server->send_notification( std::move( notification ) );
 }
 
+
+void dds_device_server::on_control_message_received()
+{
+    topics::flexible_msg data;
+    eprosima::fastdds::dds::SampleInfo info;
+    while ( topics::flexible_msg::take_next( *_control_reader, &data, &info ) )
+    {
+        if ( !data.is_valid() )
+            continue;
+
+        _control_dispatcher.invoke( [&]( dispatcher::cancellable_timer ) { handle_control_message( data ); } );

```

Comment:
Need to be really careful here:
`data` is local and will likely go out of scope by the time the dispatcher handles things. A crash is very likely eventually. The `&` will pass it to the lambda by ref which is no good -- we need to move it to the lambda.

Reply:
`invoke` is moving when enqueuing

Reply:
It moves the lambda, with all its contexts. But if the lambda contains a reference, the moved-to lambda will still contain the same reference to the original object, which sits on the stack.

Reply:
Apparently, moving captures is not supported by C++11.
Using `std::bind` workaround [See here for details](https://stackoverflow.com/a/12744621)